coral-trino/src/test/java/com/linkedin/coral/trino/rel2trino/RelToTrinoConverterTest.java

```diff
@@ -356,38 +356,53 @@ public void testTruncate() {
     testConversion(sql2, expectedSql2);
   }
 
+  @Test
+  public void testSubStr1() {
+    String sql = "SELECT SUBSTR(scol, 0) FROM test.tableOne";
+    String expectedSql =
+        "SELECT \"substr\"(\"tableone\".\"scol\", \"greatest\"(0, 1))\n" + "FROM \"test\".\"tableone\" AS \"tableone\"";
+    testConversion(sql, expectedSql);
+  }
+
   @Test
   public void testSubStr2() {
     String sql = "SELECT SUBSTR(scol, 1) FROM test.tableOne";
     String expectedSql =
-        "SELECT \"substr\"(\"tableone\".\"scol\", 1 + 1)\n" + "FROM \"test\".\"tableone\" AS \"tableone\"";
+        "SELECT \"substr\"(\"tableone\".\"scol\", \"greatest\"(1, 1))\n" + "FROM \"test\".\"tableone\" AS \"tableone\"";
     testConversion(sql, expectedSql);
   }
 
   @Test
   public void testSubString2() {
     String sql = "SELECT SUBSTRING(scol, 1) FROM test.tableOne";
     String expectedSql =
-        "SELECT \"substr\"(\"tableone\".\"scol\", 1 + 1)\n" + "FROM \"test\".\"tableone\" AS \"tableone\"";
+        "SELECT \"substr\"(\"tableone\".\"scol\", \"greatest\"(1, 1))\n" + "FROM \"test\".\"tableone\" AS \"tableone\"";
     testConversion(sql, expectedSql);
   }
 
   @Test
   public void testSubStr3() {
     String sql = "SELECT SUBSTR(scol, icol, 3) FROM test.tableOne";
-    String expectedSql = "SELECT \"substr\"(\"tableone\".\"scol\", \"tableone\".\"icol\" + 1, 3)\n"
+    String expectedSql = "SELECT \"substr\"(\"tableone\".\"scol\", \"greatest\"(\"tableone\".\"icol\", 1), 3)\n"
         + "FROM \"test\".\"tableone\" AS \"tableone\"";
     testConversion(sql, expectedSql);
   }
 
   @Test
   public void testSubString3() {
     String sql = "SELECT SUBSTRING(scol, icol, 3) FROM test.tableOne";
-    String expectedSql = "SELECT \"substr\"(\"tableone\".\"scol\", \"tableone\".\"icol\" + 1, 3)\n"
+    String expectedSql = "SELECT \"substr\"(\"tableone\".\"scol\", \"greatest\"(\"tableone\".\"icol\", 1), 3)\n"
         + "FROM \"test\".\"tableone\" AS \"tableone\"";
     testConversion(sql, expectedSql);
   }
 
+  @Test
+  public void testSubString4() {
```

Comment:
I didn't find `testSubString1`, let's make `testSubStringx` and `testSubStrx`  test the similar SQLs?

Reply:
fixed, ty!